fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/ChildrenRdfContext.java

```diff
@@ -62,29 +57,20 @@ public ChildrenRdfContext(final FedoraResource resource,
             LOGGER.trace("Found children of this resource: {}", resource.getPath());
 
             // Count the number of children
-            final Iterator<FedoraResource> childrenCounter = resource().getChildren();
-            concat(of(createNumChildrenTriple(Iterators.size(childrenCounter))));
-            concat(iteratorToStream(resource().getChildren()).map(child2triple));
-
+            concat(of(createNumChildrenTriple(resource().getChildren().count())));
+            concat(resource().getChildren().peek(x -> LOGGER.info("Creating triple for child node: {}", x))
+                    .map(x -> create(subject(), CONTAINS.asNode(), x instanceof NonRdfSourceDescription ?
```

Comment:
Maybe use `child` instead of `x`? I hate the variable name `x`, even though I something use it.